### ISSUE TITLE
Azure: multi-tenant application to access customer data (again)

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#579]
In GitLab by @infinitewarp on May 14, 2019, 14:32

This is a followup to https://gitlab.com/cloudigrade/cloudigrade/issues/514 since half a year has passed and Azure things may have changed in the interim.

Review and define instructions with a prototype script for connecting to and performing some operation (like "list all currently running instances") in a customer Azure account from our cloudigrade Azure account via API calls.

Note that we don't care today about restricting permissions. We just need to be able to get in and do things in others' accounts.

Time box: 1 week.

**Comment by katherine-black** [comment]
In GitLab by @katherine-black on May 28, 2019, 09:22

Unfortunately this spike did not really yield anything new. I spent quite a bit of time investigating alternative ways to get permissions needed for our application, avoiding needing the graph API permissions, and relying strictly on the `user_impersonate` permissions available through the azure management api. 

Unfortunately most of the investigation ended in a dead end, seemingly requiring me to deploy an API service before I could actually begin to impersonate a user. Considering it had to be deployed in a users account, and there is :money_with_wings: involved, it seemed like a no go.

So, bad news? Nothing changed. Good news? Nothing changed.

I'll link Bihan's comment from the time she did the spike as all of that information is still current: https://gitlab.com/cloudigrade/cloudigrade/issues/514#note_127215591

And I'll attach my own simplified notebook where I verified that the process noted before still functions. 
[573-multitenant-again.ipynb](/uploads/430a920793abb126271474979b2dca63/573-multitenant-again.ipynb)
